### PR TITLE
`logging`: log equiv. of `LOG_LEVEL=debug` to files in `~/.fly/logs`

### PR DESCRIPTION
**Pull request by alichay**
### Change Summary

What and Why:

To make troubleshooting easier and issues more reproducible, we'll keep logs for the last 10 (this is a totally arbitrary number) runs of flyctl. These logs are the equivalent of `LOG_LEVEL=debug`, although they won't contain the actual output - only the debug logs.

How:

A lot of cleanup around logging and the `terminal` package. We had like four reimplementations of this logging system scattered around flyctl lmao, this consolidates most of it.

Related to: [internal discussion](https://flyio.discourse.team/t/flyctl-diagnostics-pipeline/3403)

---

### Documentation

- [ ] Fresh Produce
- [ ] In superfly/docs, or asked for help from docs team
- [X] n/a (this is more a tool for support rather than something user-facing)

----

Sorry about the mega-commit! I kind of dug myself into a hole, and had to rewrite things to get out of it - in the process I just totally lost any hope of a reasonable commit history.

### CODE REVIEW

File: internal/logger/global_logfile.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package logger
+
+import (
+	"bufio"
+
+	"github.com/superfly/flyctl/internal/logger/logfile"
+)
+
+var globalLogFile = logFile{
+	destroyed: true,
```
Comment (dangra) [inline internal/logger/global_logfile.go:10]: does it need a mutex ?

**Review by dangra — APPROVED** [review]

File: internal/logger/global_logfile.go (hunk 1 above)
Comment (alichay) [reply internal/logger/global_logfile.go:10]: I don't think go mutexes require initialization, right?

Comment (dangra) [reply internal/logger/global_logfile.go:10]: sorry, I blindly asked.

The question was towards protecting the global variable from changing when InitLogFile or CloseLogFile are called.

Comment (alichay) [reply internal/logger/global_logfile.go:10]: ahhh ok :)
I think it _should_ be fine?

**Pull request merged**